### PR TITLE
refactor: a bunch of trivial-ish headers improvements

### PR DESCRIPTION
**Pull request by knst**
## Issue being fixed or feature implemented
Split from https://github.com/dashpay/dash/pull/6375 and added some extra

## What was done?
Tidy up headers of evo/ and llmq/:
 - removed unused includes and added missing
 - removed unused forward declarations and added missing
 - removed helper GetCoinbaseTx, moved helper IsQuorumDKGEnabled from header to implementation

## How Has This Been Tested?
Run unit and functional tests

Compilation time _in theory_ should be improved, in really it shaved just 2-5 seconds from 5 minutes by my benchmark.

## Breaking Changes
N/A

## Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

File: src/evo/cbtx.cpp [hunk 1]
```diff
@@ -448,7 +449,7 @@ std::string CCbTx::ToString() const
         creditPoolBalance / COIN, creditPoolBalance % COIN);
 }
 
-std::optional<CCbTx> GetCoinbaseTx(const CBlockIndex* pindex)
+std::optional<std::pair<CBLSSignature, uint32_t>> GetNonNullCoinbaseChainlock(const CBlockIndex* pindex)
```
Comment (UdjinM6) [inline src/evo/cbtx.cpp:452]: 6697ad1202d745521ee6b8ec0e931811ae522519: commit title should say smth like "drop GetCoinbaseTx", not "make GetCoinbaseTx static"

File: src/evo/assetlocktx.h [hunk 2]
```diff
@@ -17,6 +17,7 @@
 class BlockManager;
 class CBlockIndex;
 class CRangesSet;
+class CTransaction;
```
Comment (UdjinM6) [inline src/evo/assetlocktx.h:20]: 0c023f0a49d54b03bcb170a6b21bb126b85eb288: nit: `#include <primitives/transaction.h>` should have this covered already
```suggestion
```

**Review by UdjinM6 — APPROVED** [review]
utACK e9387eef69f66bee6419575933022963b2b85fb0

**Review by PastaPastaPasta — APPROVED** [review]
utACK e9387eef69f66bee6419575933022963b2b85fb0

**Pull request merged**